server/src/main/java/org/elasticsearch/transport/OutboundMessage.java

```diff
@@ -117,17 +116,12 @@ private static byte setStatus(boolean compress, boolean isHandshake, Writeable m
 
     static class Response extends OutboundMessage {
 
-        private final Set<String> features;
-
-        Response(ThreadContext threadContext, Set<String> features, Writeable message, Version version, long requestId,
-                 boolean isHandshake, boolean compress) {
+        Response(ThreadContext threadContext, Writeable message, Version version, long requestId, boolean isHandshake, boolean compress) {
             super(threadContext, version, setStatus(compress, isHandshake, message), requestId, message);
-            this.features = features;
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.setFeatures(features);
```

Comment:
This was (and now is even more) a weird `writeTo` that doesn't actually write anything and probably worth changing the abstractions around serializing this thing around a bit to make things cleaner.